File: CS/WinWebSolution.Module/MyNonPersistentDomainObjectNavigationController.cs
using System;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.SystemModule;
using DevExpress.Xpo;

namespace WinWebSolution.Module {
    public class MyNonPersistentDomainObjectNavigationController: ShowNonPersistentObjectDetailViewFromNavigationControllerBase<MyNonPersistentDomainObject> {
        protected override void CustomizeNonPersistentObject(MyNonPersistentDomainObject obj) {
            base.CustomizeNonPersistentObject(obj);
            obj.Name = "MyNonPersistentDomainObject";
            //Dennis: do more customizations here
        }
        protected override void CustomizeShowViewParameters(ShowViewParameters parameters) {
            base.CustomizeShowViewParameters(parameters);
            //Dennis: do more customizations here
        }
    }
}

File: VB/WinWebSolution.Module/MyNonPersistentDomainObjectNavigationController.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.SystemModule
Imports DevExpress.Xpo

Namespace WinWebSolution.Module
	Public Class MyNonPersistentDomainObjectNavigationController
		Inherits ShowNonPersistentObjectDetailViewFromNavigationControllerBase(Of MyNonPersistentDomainObject)
		Protected Overrides Sub CustomizeNonPersistentObject(ByVal obj As MyNonPersistentDomainObject)
			MyBase.CustomizeNonPersistentObject(obj)
			obj.Name = "MyNonPersistentDomainObject"
			'Dennis: do more customizations here
		End Sub
		Protected Overrides Sub CustomizeShowViewParameters(ByVal parameters As ShowViewParameters)
			MyBase.CustomizeShowViewParameters(parameters)
			'Dennis: do more customizations here
		End Sub
	End Class
End Namespace